core/controllers/story_viewer.py

```diff
@@ -244,6 +245,8 @@ def post(self, story_id, node_id):
         else:
             learner_progress_services.mark_topic_as_learnt(
                 self.user_id, topic.id)
+            learner_goals_services.remove_topic_from_learn(
```

Comment:
Maybe `remove_topic_from_learn_goal`? That gives more context as to from where you are removing this topic.

Reply:
Done.